README.md
# Seeker
This application is a data browser. It allows people to search data in a database (for now VoNZ one) following some parameters and after that download selected datum in their own Owncloud.

For now the parameters which can be selected are gender, age, location, english profile, maori rofile, time in new zealand,and type of recording.

# Installation
Download the seeker repository and place it in **owncloud/apps/**

Warning : The name of the repository HAVE TO be seeker !!

## Publish to App Store

First get an account for the [App Store](http://apps.owncloud.com/) then run:

    make appstore_package

The archive is located in build/artifacts/appstore and can then be uploaded to the App Store.

## Running tests
After [Installing PHPUnit](http://phpunit.de/getting-started.html) run:

    phpunit -c phpunit.xml
    
 ## More

To have more informations on how you can manage this application, let's see the [Owncloud Developer Manual](https://doc.owncloud.org/server/latest/developer_manual/app/index.html)...
    


CHANGELOG.md
owncloud-seeker (0.0.1)
* First release

tests/integration/AppTest.php
<?php
/**
 * ownCloud - seeker
 *
 * This file is licensed under the Affero General Public License version 3 or
 * later. See the COPYING file.
 *
 * @author Daugieras <adau828@aucklanduni.ac.nz>
 * @copyright Daugieras 2017
 */

use OCP\AppFramework\App;
use Test\TestCase;


/**
 * This test shows how to make a small Integration Test. Query your class
 * directly from the container, only pass in mocks if needed and run your tests
 * against the database
 */
class AppTest extends TestCase {

    private $container;

    public function setUp() {
        parent::setUp();
        $app = new App('seeker');
        $this->container = $app->getContainer();
    }

    public function testAppInstalled() {
        $appManager = $this->container->query('OCP\App\IAppManager');
        $this->assertTrue($appManager->isInstalled('seeker'));
    }

}

tests/unit/controller/PageControllerTest.php
<?php
/**
 * ownCloud - seeker
 *
 * This file is licensed under the Affero General Public License version 3 or
 * later. See the COPYING file.
 *
 * @author Daugieras <adau828@aucklanduni.ac.nz>
 * @copyright Daugieras 2017
 */

namespace OCA\Seeker\Controller;

use PHPUnit_Framework_TestCase;

use OCP\AppFramework\Http\TemplateResponse;


class PageControllerTest extends PHPUnit_Framework_TestCase {

	private $controller;
	private $userId = 'john';

	public function setUp() {
		$request = $this->getMockBuilder('OCP\IRequest')->getMock();

		$this->controller = new PageController(
			'seeker', $request, $this->userId
		);
	}


	public function testIndex() {
		$result = $this->controller->index();

		$this->assertEquals(['user' => 'john'], $result->getParams());
		$this->assertEquals('main', $result->getTemplateName());
		$this->assertTrue($result instanceof TemplateResponse);
	}


	public function testEcho() {
		$result = $this->controller->doEcho('hi');
		$this->assertEquals(['echo' => 'hi'], $result->getData());
	}


}

js/script.js
/**
 * ownCloud - seeker
 *
 * This file is licensed under the Affero General Public License version 3 or
 * later. See the COPYING file.
 *
 * @author Daugieras <adau828@aucklanduni.ac.nz>
 * @copyright Daugieras 2017
 */

(function ($, OC) {

	$(document).ready(function () {

		var choice;
		var age;
		var nztime;
		var users;
		var fileName;
		var fileId;
		var fileToDownload;
		var nb =0;
		var k =0;
		var echo = document.getElementById('echo-result');

		//initialisation
		document.getElementById("All").checked = true;
		findFile();


		$('#find').click(function () {
			var url = OC.generateUrl('/apps/seeker/owner'); 
			RecordingType();
			var data = {
				users :users,
				type: choice
			};

			$.post(url, data).success(function (response) { //call fonction getOwner in pagecontroller.php
				fileName = response.echo;
				fileId = response.echo2;
				$('#tableau').remove(); //Delete old table before create a new one with new data
				Display();		//Create table with new data
				$('#echo-result').removeClass('hidden');
			});
		});

		$('#download').click(function () {
			var url = OC.generateUrl('/apps/seeker/download');
			FileChoice(); 		//update fileToDownload

			var data = {

				fileId: fileToDownload,
				folderName :document.getElementById('folder').value,
			};

			info.innerHTML = "In process";

			$.post(url, data)	//call fonction download in pagecontroller.php	 
			  .done(function() {
				  document.getElementById('info').style.color = 'green';
				  $(info).text("Success!");
			  })
			  .fail(function() {
				  document.getElementById('info').style.color = 'red';
				$(info).text("Error!");
			  })
			  .always(function() {
			  });
		});

		//function carry out each time that a parameter changed, initialize variables and update the list of users who fit the parameters
		function findFile (){ 
			var url = OC.generateUrl('/apps/seeker/user');
			AgeBand();
			NzLiveBand();

			var data = {
				gender:document.getElementById('gender').value,
				location: document.getElementById('addr').value,
				old :age,
				english: document.getElementById('eprofil').value,
				maori: document.getElementById('mprofil').value,
				nzlive:nztime
			};
				result.innerHTML = " users found !";
			$.post(url, data).success(function (response) { 	//call getUser download in pagecontroller.php	 
				users = response.echo;
				result.innerHTML = users.length+" users found !";

			});

			//initialialization again 
			$('#echo-result').addClass('hidden');
			document.getElementById('info').style.color = 'black';
			info.innerHTML = "";
		}


		function RecordingType (){
			choice = [0,0,0,0,0];

			if(document.getElementById('All').checked == true){
			choice[0] =1;
			}

			if(document.getElementById('Word').checked == true){
			choice[1] =1;
			}

			if(document.getElementById('ListOfWord').checked == true){
			choice[2] =1;
			}

			if(document.getElementById('ShortSentence').checked == true){
			choice[3] =1;
			}

			if(document.getElementById('Sentence').checked == true){
			choice[4] =1;
			}
		}

		function AgeBand () {
			switch (document.getElementById('age').value)
			{
			case 'All': age = 'All';	
					break;
			case '16-25': age = 16;	
					break;
			case '26-35': age = 26;	
					break;
			case '36-45': age = 36;	
					break;
			case '46-55': age = 46;	
					break;
			case '56-65': age = 56;	
					break;
			case '66-75': age = 66;	
					break;
			case '76-85': age = 76;	
					break;
			case '+85': age = 86;	
					break;
			}
		}
		
		function NzLiveBand () {
			switch (document.getElementById('nztime').value)
			{
			case 'All': nztime = 'All';	
					break;
			case '0-5': nztime = 0;	
					break;
			case '6-10': nztime = 6;	
					break;
			case '11-20': nztime = 11;	
					break;
			case '20+': nztime = 20;	
					break;
			}
		}

		function Display(){

			var tab = document.createElement('table');
			tab.id = 'tableau';
			nb =(fileName.length);

			//Fill the table only if fileName is not empty
			if(nb !== 0){ 
				fileName[nb]='All';

				for (row = 0; row < nb+1; row++){
					var tr = document.createElement('tr');
				   	
				      	td = document.createElement('td');
				      	td2 = document.createElement('td');

					var tn = document.createElement('input') ;
					tn.setAttribute('type','checkbox');
					tn.setAttribute('id','tr'+row);
					var tn2 = document.createTextNode(fileName[nb-row]); //use nb-row to have "All" first and so always with the id : tr0
	
				      	td.appendChild(tn);
				      	td2.appendChild(tn2);

					tr.appendChild(td);
					tr.appendChild(td2);

					tab.appendChild(tr);
				}
			}

			echo.appendChild(tab);
			$('#tableau').insertBefore('#part2'); //insert before download button 
		}
		
		//When checkbox 'All' is activate every other checkbox will be activate too
		function check(){
			if(document.getElementById('tr0').checked == true){
				for (i = 0; i < nb+1; i++){document.getElementById('tr'+i).checked = true;}
			}

		}
		

		function FileChoice(){

			fileToDownload = [];
			k=0;

			for (i = 1; i < nb+1; i++){
				if(document.getElementById('tr'+i).checked == true){
					fileToDownload[k] =fileId[nb-i]; //be careful of the index du to the inversion of index when filling the table (see Display function)
					k++;
				}
				
			}
		}
		
		var $activityNotifications = $('#seeker');
		//search files when users add informations from input or select
		$activityNotifications.find('select').change(findFile);
		$activityNotifications.find('#recordingtype').change(findFile);
		$activityNotifications.find('#echo-result').change(check);

	});

})(jQuery, OC);


appinfo/app.php
<?php
/**
 * ownCloud - seeker
 *
 * This file is licensed under the Affero General Public License version 3 or
 * later. See the COPYING file.
 *
 * @author Daugieras <adau828@aucklanduni.ac.nz>
 * @copyright Daugieras 2017
 */

namespace OCA\Seeker\AppInfo;

use OCP\AppFramework\App;

$app = new App('seeker');
$container = $app->getContainer();

$container->query('OCP\INavigationManager')->add(function () use ($container) {
	$urlGenerator = $container->query('OCP\IURLGenerator');
	$l10n = $container->query('OCP\IL10N');
	return [
		// the string under which your app will be referenced in owncloud
		'id' => 'seeker',

		// sorting weight for the navigation. The higher the number, the higher
		// will it be listed in the navigation
		'order' => 10,

		// the route that will be shown on startup
		'href' => $urlGenerator->linkToRoute('seeker.page.index'),

		// the icon that will be shown in the navigation
		// this file needs to exist in img/
		'icon' => $urlGenerator->imagePath('seeker', 'researcher.svg'),

		// the title of your application. This will be used in the
		// navigation or on the settings page of your app
		'name' => $l10n->t('Seeker'),
	];
});


appinfo/routes.php
<?php
/**
 * ownCloud - seeker
 *
 * This file is licensed under the Affero General Public License version 3 or
 * later. See the COPYING file.
 *
 * @author Daugieras <adau828@aucklanduni.ac.nz>
 * @copyright Daugieras 2017
 */

/**
 * Create your routes in here. The name is the lowercase name of the controller
 * without the controller part, the stuff after the hash is the method.
 * e.g. page#index -> OCA\Seeker\Controller\PageController->index()
 *
 * The controller class has to be registered in the application.php file since
 * it's instantiated in there
 */
return [
    'routes' => [
	   ['name' => 'page#index', 'url' => '/', 'verb' => 'GET'],
	   ['name' => 'page#getUser', 'url' => '/user', 'verb' => 'POST'],
	   ['name' => 'page#getOwner', 'url' => '/owner', 'verb' => 'POST'],
	   ['name' => 'page#download', 'url' => '/download', 'verb' => 'POST'],
    ]
];


controller/pagecontroller.php
<?php
/**
 * ownCloud - seeker
 *
 * This file is licensed under the Affero General Public License version 3 or
 * later. See the COPYING file.
 *
 * @author Daugieras <adau828@aucklanduni.ac.nz>
 * @copyright Daugieras 2017
 */

namespace OCA\Seeker\Controller;

use OCP\IRequest;
use OCP\AppFramework\Http\TemplateResponse;
use OCP\AppFramework\Http\DataResponse;
use OCP\AppFramework\Controller;
use OCP\IUserSession;
use OCP\IConfig;
use OCP\Files\FileInfo;
use OCP\Files\Folder;
use OCP\Files\IRootFolder;
use OCP\Files\Node;

class PageController extends Controller {


	private $userId;
	private $userSession;
	private $config;

	public function __construct($AppName, IRequest $request, IUserSession $userSession, $UserId, IConfig $config){
		parent::__construct($AppName, $request);
		$this->userId = $UserId;
		$this->userSession = $userSession;
		$this->config = $config;
	}

	/**
	 * CAUTION: the @Stuff turns off security checks; for this page no admin is
	 *          required and no CSRF check. If you don't know what CSRF is, read
	 *          it up in the docs or you might create a security hole. This is
	 *          basically the only required method to add this exemption, don't
	 *          add it to any other method if you don't exactly know what it does
	 *
	 * @NoAdminRequired
	 * @NoCSRFRequired
	 */
	public function index() {
		$params = ['user' => $this->userId];
		return new TemplateResponse('seeker', 'main', $params);  // templates/main.php
	}

	/**
	 * Simply method that posts back users who fit to the selected parameters .
	 * @NoAdminRequired
	 */

	public function getUser($gender,$location,$old,$english,$maori,$nzlive){

		$users =array();		

		if($gender === 'both')
		{
			$users = array_merge($this->config->getUsersForUserValue('background','gender','male'),$this->config->getUsersForUserValue('background','gender','female'));
		}
		else{$users = $this->config->getUsersForUserValue('background','gender',$gender);}

		$size = count($users);

		if($location !== 'All')
		{	
			for($i=0; $i<$size; $i++)
			{
			$addr = $this->config->getUserValue($users[($size-$i-1)],'background','addr');
			if($addr !== $location) {array_splice($users,$size-$i-1,1);}
			}
		}

		$size = count($users);

		if($old !== 'All')
		{	
			for($i=0; $i<$size; $i++)
			{
			$age = $this->config->getUserValue($users[($size-$i-1)],'background','age');
			if(($age<$old || $age>$old+9) && ($old !== '86')) {array_splice($users,$size-$i-1,1);}
			if($age<$old && $old === '86') {array_splice($users,$size-$i-1,1);}
			}
		}

		$size = count($users);

		if($english !== 'none')
		{
			for($i=0; $i<$size; $i++)
			{
				$en = $this->config->getUserValue($users[($size-$i-1)],'background','eprofil');
	
				if($english === 'both' && $en === 'other') {array_splice($users,$size-$i-1,1);}
				if($english === 'L2' && $en !== 'L2') {array_splice($users,$size-$i-1,1);}
				if(($english === 'L1(NZEnglish)' || $english === 'L1(All)') && $en !== 'L1') {array_splice($users,$size-$i-1,1);}
				if($english === 'L1(NZEnglish)' && $en === 'L1') 
				{
				$type = $this->config->getUserValue($users[($size-$i-1)],'background','etype');
					if($type !== 'English(NewZeland)'){array_splice($users,$size-$i-1,1);}
				}
			}
		}

		$size = count($users);

		if($maori !== 'none')
		{
			for($i=0; $i<$size; $i++)
			{
				$ma = $this->config->getUserValue($users[($size-$i-1)],'background','mprofil');
	
				if($maori === 'both' && $ma === 'other') {array_splice($users,$size-$i-1,1);}
				if($maori === 'L2' && $ma !== 'L2') {array_splice($users,$size-$i-1,1);}
				if($maori === 'L1' && $ma !== 'L1') {array_splice($users,$size-$i-1,1);}
	
			}
		}

		$size = count($users);

		if($nzlive !== 'All')
		{	
			for($i=0; $i<$size; $i++)
			{
			$nztime = $this->config->getUserValue($users[($size-$i-1)],'background','nztime');
			$test[$i]=$nztime;
			if(($nztime<$nzlive || $nztime>$nzlive+5) && ($nzlive === '0')) {array_splice($users,$size-$i-1,1);}
			if(($nztime<$nzlive || $nztime>$nzlive+4) && ($nzlive === '6')) {array_splice($users,$size-$i-1,1);}
			if(($nztime<$nzlive || $nztime>$nzlive+9) && ($nzlive === '11')) {array_splice($users,$size-$i-1,1);}
			if($nztime<$nzlive && $nzlive === '20') {array_splice($users,$size-$i-1,1);}
			}
		}
		
		return new DataResponse(['echo' => $users]);
	}


	/**
	 * Simply method that posts back files which belong to the folder selected and created by users selected 
	 * @NoAdminRequired
	 */
	public function getOwner($type,$users) {

		$file_array =array();
		$fileName = array();
		$fileId = array();
		$owner =array();
		$folderType = array();
		$folderChoice = array();
		$l =0;
		$folder = \OC::$server->getUserFolder($userId); //get the root folder of the user 
		$sizeusers = count($users);

		//search in the root folder each database folder and assign them to one specific index of folderChoice
		$folderType = $folder->search('DataBase VoNZ word');
		$folderChoice[0] = $folderType[0];
		$folderType = $folder->search('DataBase VoNZ list_word');
		$folderChoice[1] = $folderType[0];
		$folderType = $folder->search('DataBase VoNZ short_sentence');
		$folderChoice[2] = $folderType[0];
		$folderType = $folder->search('DataBase VoNZ sentence');
		$folderChoice[3] = $folderType[0];

		//for each database folder
		for($i=0; $i<4; $i++)
		{	
			//if the folder is selected in type
			if ($type[($i+1)]=== '1' || $type[0] === '1'){

				//take all the files in this folder
				$file_array = $folderChoice[$i]->getDirectoryListing();
				$sizefile = count($file_array);

				//for each file in the folder
				for($j=0; $j<$sizefile; $j++)
				{	
						//for each user selected
						for($k=0; $k<$sizeusers; $k++)
						{
							//look if the file was created by the user
							$owner[$j] = strtok($file_array[$j]->getName(),'_');
							if( $owner[$j] === $users[$k])
							{
								$fileName[$l] = $file_array[$j]->getName();	//if yes add the name file to the list of file : fileName
								$fileId[$l] = $file_array[$j]->getId();		//and the file id to the list :fileId
								$l++; //to have successive index in fileName 
							}		
						}
				}
			}
		}

		return new DataResponse(['echo' => $fileName, 'echo2' =>$fileId]); //send the list of files which match 
	}

	/**
	 * Simply method that download fileId in the folderName
	 * @NoAdminRequired
	 */
	public function download($fileId, $folderName) {

		$folder = \OC::$server->getUserFolder($userId); //get the root folder of the user 
		$size = count($fileId);
		$uid = $this->userSession->getUser()->getUID(); //collect the user ID
		
		if($folderName === ''){$folderName = 'Study_1';} //name by default = Study_1

		if($folder->nodeExists('/'.$folderName.'/') === false) {$folder->newFolder('/'.$folderName.'/');} //if the folder does not exist yet, create it 

		//For each file copy it on the folder selected
		for($i=0; $i<$size; $i++)
		{	
			$file = $folder->getById($fileId[$i]);
			$path = '/'.$uid.'/files/'.$folderName.'/'.$file[0]->getName();
			$file[0]->copy($path);			
		}
		


		return new DataResponse();
	}

}


templates/part.content.php
<div id = seeker>
<br>
<br>
<a target="_blank">
	<img src="<?php print_unescaped(OCP\Util::imagePath('seeker', 'researcher.jpg')); ?>" style="height:150px"/>
</a>

<p>Welcome <?php p($_['user']) ?></p>
	<br>
<p>Choose your parameters :</p>
	<br>
<label>Gender</label>

	<select class="form-control" id="gender" name="gender">
		<option value="both" <?php if (($_['gender']) == 'both') echo ' selected="selected"'; ?>>both</option>
		<option value="male" <?php if (($_['gender']) == 'male') echo ' selected="selected"'; ?>>male</option>
		<option value="female" <?php if (($_['gender']) == 'female') echo ' selected="selected"'; ?>>female</option>
	</select>

<label>Location</label>

	<select class="form-control" id="addr" name="addr">
		<option value="All" <?php if (($_['addr']) == 'All') echo ' selected="selected"'; ?>>All</option>
		<option value="Northland" <?php if (($_['addr']) == 'Northland') echo ' selected="selected"'; ?>>Northland</option>
		<option value="Auckland" <?php if (($_['addr']) == 'Auckland') echo ' selected="selected"'; ?>>Auckland</option>
		<option value="Waikato" <?php if (($_['addr']) == 'Waikato') echo ' selected="selected"'; ?>>Waikato</option>
		<option value="BayOfPlenty" <?php if (($_['addr']) == 'BayOfPlenty') echo ' selected="selected"'; ?>>Bay of Plenty</option>
		<option value="Gisborne" <?php if (($_['addr']) == 'Gisborne') echo ' selected="selected"'; ?>>Gisborne</option>
		<option value="HawkeBay" <?php if (($_['addr']) == 'HawkeBay') echo ' selected="selected"'; ?>>Hawke Bay</option>
		<option value="Taranaki" <?php if (($_['addr']) == 'Taranaki') echo ' selected="selected"'; ?>>Taranaki</option>
		<option value="Manawatu-Wanganui" <?php if (($_['addr']) == 'Manawatu-Wanganui') echo ' selected="selected"'; ?>>Manawatu-Wanganui</option>
		<option value="Wellington" <?php if (($_['addr']) == 'Wellington') echo ' selected="selected"'; ?>>Wellington</option>
		<option value="Marlborough" <?php if (($_['addr']) == 'Marlborough') echo ' selected="selected"'; ?>>Marlborough</option>
		<option value="Tasman" <?php if (($_['addr']) == 'Tasman') echo ' selected="selected"'; ?>>Tasman</option>
		<option value="WestCoast" <?php if (($_['addr']) == 'WestCoast') echo ' selected="selected"'; ?>>West Coast</option>
		<option value="Canterbury" <?php if (($_['addr']) == 'Canterbury') echo ' selected="selected"'; ?>>Canterbury</option>
		<option value="Otago" <?php if (($_['addr']) == 'Otago') echo ' selected="selected"'; ?>>Otago</option>
		<option value="Southland" <?php if (($_['addr']) == 'Southland') echo ' selected="selected"'; ?>>Southland</option>
	</select>

<label>Age</label>

	<select class="form-control" id="age" name="age">
		<option value="All" <?php if (($_['age']) == 'All') echo ' selected="selected"'; ?>>All</option>
		<option value="16-25" <?php if (($_['age']) == '16-25') echo ' selected="selected"'; ?>>16-25</option>
		<option value="26-35" <?php if (($_['age']) == '26-35') echo ' selected="selected"'; ?>>26-35</option>
		<option value="36-45" <?php if (($_['age']) == '36-45') echo ' selected="selected"'; ?>>36-45</option>
		<option value="46-55" <?php if (($_['age']) == '46-55') echo ' selected="selected"'; ?>>46-55</option>
		<option value="56-65" <?php if (($_['age']) == '56-65') echo ' selected="selected"'; ?>>56-65</option>
		<option value="66-75" <?php if (($_['age']) == '66-75') echo ' selected="selected"'; ?>>66-75</option>
		<option value="76-85" <?php if (($_['age']) == '76-85') echo ' selected="selected"'; ?>>76-85</option>
		<option value="+85" <?php if (($_['age']) == '+85') echo ' selected="selected"'; ?>>+85</option>
	</select>

<label>English profile</label>

	<select class="form-control" id="eprofil" name="eprofil">
		<option value="none" <?php if (($_['eprofil']) == 'none') echo ' selected="selected"'; ?>>none</option>
		<option value="both" <?php if (($_['eprofil']) == 'both') echo ' selected="selected"'; ?>>both</option>
		<option value="L1(All)" <?php if (($_['eprofil']) == 'L1(All)') echo ' selected="selected"'; ?>>L1 (All)</option>
		<option value="L1(NZEnglish)" <?php if (($_['eprofil']) == 'L1(NZEnglish)') echo ' selected="selected"'; ?>>L1 (NZ English)</option>
		<option value="L2" <?php if (($_['eprofil']) == 'L2') echo ' selected="selected"'; ?>>L2</option>
	</select>

<label>Maori profile</label>

	<select class="form-control" id="mprofil" name="mprofil">
		<option value="none" <?php if (($_['mprofil']) == 'none') echo ' selected="selected"'; ?>>none</option>
		<option value="both" <?php if (($_['mprofil']) == 'both') echo ' selected="selected"'; ?>>both</option>
		<option value="L1" <?php if (($_['mprofil']) == 'L1') echo ' selected="selected"'; ?>>L1</option>
		<option value="L2" <?php if (($_['mprofil']) == 'L2') echo ' selected="selected"'; ?>>L2</option>
	</select>

<label>Nb years in NZ</label>

	<select class="form-control" id="nztime" name="nztime">
		<option value="All" <?php if (($_['nztime']) == 'All') echo ' selected="selected"'; ?>>All</option>
		<option value="0-5" <?php if (($_['nztime']) == '0-5') echo ' selected="selected"'; ?>>0-5</option>
		<option value="6-10" <?php if (($_['nztime']) == '6-10') echo ' selected="selected"'; ?>>6-10</option>
		<option value="11-20" <?php if (($_['nztime']) == '11-20') echo ' selected="selected"'; ?>>11-20</option>
		<option value="20+" <?php if (($_['nztime']) == '20+') echo ' selected="selected"'; ?>>20+</option>
	</select>
<br>
<br>
<p> Recording type : </p> 

<div id ='recordingtype'>	
	<input id = "All" type = checkbox> <label>All</label>
	<input id = "Word" type = checkbox> <label>Word</label>
	<input id = "ListOfWord" type = checkbox> <label>List of Word</label>
	<input id = "ShortSentence" type = checkbox> <label>Short Sentence</label>
	<input id = "Sentence" type = checkbox> <label>Sentence</label>
</div>
<br>
<br>
	<label id ="result" style ="color:red;"></label>
	<button id ="find">Show files</button>
<br>
<br>

<div id="echo-result">
	<div id ='part2'>
<br>
	<button id ="download">Download</button>
    	<label for="folder" class="col-sm-2 control-label">in</label>
	<input type="text" class="form-control update" id="folder" name="folder" placeholder="Study_1"/>
	<label id ="info"></label>
	</div>
</div>
